client/app/scripts/superdesk/services/modalService.js

```diff
@@ -43,5 +43,53 @@ define(['angular'], function(angular) {
                 return delay.promise;
             };
 
+        }])
+        .directive('sdModal', ['$document', function($document) {
+
+            return {
+                template: [
+                    '<div class="modal fade hide">',
+                    '<div class="modal-dialog" ng-if="isModalOpen"><div class="modal-content" ng-transclude></div></div>',
+                    '</div>'].join(''),
+                transclude: true,
+                link: function(scope, element, attrs) {
+
+                    var content, _initialized = false;
+                    var trigger = attrs.open;
```

Comment:
Well, using isolated scopes is actually a _recommended_ for reusable components (for the exact reason I mentioned above).

By "extendable" you probably mean "configurable"? General-prupose components are easily configurable, it's done using the extra attributes on the DOM element a directive is used on. For example, specifying a _save()_ handler can be done like this:

<element data-onsave="&onSaveCallbackHandler()" ....>